bin/utils/s3.py

```diff
@@ -0,0 +1,150 @@
+import mimetypes
+from functools import reduce
+from dotmap import DotMap
+from urllib.parse import urlparse
+import boto3
+from boto3.s3.transfer import TransferConfig
+from botocore.exceptions import ClientError
+from checksumming_io.checksumming_io import ChecksummingBufferedReader, ChecksummingSink, S3Etag
+from .parallel_logger import logger
+
+
+class S3Location(DotMap):
+
+    def __init__(self, url):
+        urlbits = urlparse(url)
+        if urlbits.scheme != 's3':
+            raise RuntimeError("Not an S3 URL!")
+        super().__init__(Bucket=urlbits.netloc, Key=urlbits.path.lstrip('/'))
+
+    def __str__(self):
+        return f"s3://{self.Bucket}/{self.Key}"
+
+
+class S3Agent:
+
+    def __init__(self):
+        self.tx_cfg = TransferConfig(multipart_threshold=S3Etag.etag_stride,
```

Comment:
Technically, it should follow the rules in https://github.com/HumanCellAtlas/data-store/pull/288